src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -472,6 +474,123 @@ private static Expression AddResponseWritingToMethodCall(Expression methodCall,
             };
         }
 
+        private static MethodInfo GetEnumTryParseMethod()
+        {
+            var staticEnumMethods = typeof(Enum).GetMethods(BindingFlags.Public | BindingFlags.Static)
+                .Where(m => m.IsGenericMethod && m.Name == "TryParse" && m.ReturnType == typeof(bool));
+
+            foreach (var method in staticEnumMethods)
+            {
+                var tryParseParameters = method.GetParameters();
+
+                if (tryParseParameters.Length == 2 &&
+                    tryParseParameters[0].ParameterType == typeof(string) &&
+                    tryParseParameters[1].IsOut)
+                {
+                    return method;
+                }
+            }
+
+            throw new Exception("static bool System.Enum.TryParse<TEnum>(string? value, out TEnum result) does not exist!!?!?");
```

Comment:
Sorry! You're going to have to rebase this. This code has been moved to [TryParseMethodCache.cs](https://github.com/dotnet/aspnetcore/commit/fd19f92df938c42525c9f3a2f3acd13d7e7d88f2#diff-9c534f41f426c3ea25cb825e13e74bf2d639e723662625f3280629425db614bc) just today. Try not to lose the changes made along with the move like the change to the exception message toning down the punctuation. 😅

Reply:
Sure !!